### ISSUE TITLE
Add option: Fail Message on Timeout Waiting for ACK.

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#1314]
Regarding help desk ticket 12860, On Jan 21, 2010... Brendan wrote:
...  "There should be a feature called Fail Message on Timeout Waiting for ACK.  This way, people could choose their behavior because half the people want it one way, and half the people want it
the other way" ...

Note by MikeO: This is based on an issue where a change was made in version 1.8.0 so that when Mirth times out waiting for an HL7 ack it will always retry sending the message if persistent queuing is turned on, (Mirth-726).  In the case of this client, this change caused thousands of duplicate messages to flood their destination.  It looks like the destination got behind in delivering ACKs so Mirth kept sending duplicate messages which further overwhelmed the remote destination.  The destination tried playing catch-up with the result that ACKs for one message were attached to hundreds of other messages.  The end result being none of the ACKs matching up to the messages recorded by mirth. 

Imported Issue. Original Details:
Jira Issue Key: MIRTH-1314
Reporter: hdmoscm
Created: 2010-01-25T06:42:44.000-0800

**Comment by rbeckman-nextgen** [comment]
Added Queue on ACK Timeout option.

Imported Comment. Original Details:
Author: jacobb
Created: 2010-03-18T12:17:08.000-0700

**Comment by rbeckman-nextgen** [comment]
Hi Jacob,

I think that the implemented change does not solve the issue reported by MikeO.

I did a test with 2.0 RC1 and if an ack-timeout occurs, Mirth Connect keeps adding the (same) message to its queue. To my opinion it should only add the message to the queue the first time the send fails (i.e. no ack is received on time in this case).

Imported Comment. Original Details:
Author: nicovn
Created: 2010-09-03T02:55:10.000-0700

**Comment by rbeckman-nextgen** [comment]
Could you explain the issue you are seeing a little more?  Queue on ACK Timeout should continue to leave the message in the queue until an ACK is received.  If the option chosen is not to Queue on ACK Timeout, then the message should error.

Imported Comment. Original Details:
Author: jacobb
Created: 2010-09-03T11:02:56.000-0700

**Comment by rbeckman-nextgen** [comment]
Hi Jacob,

I'm sorry ... today I'm unable to reproduce the issue I had last Friday ...

The issue that I encountered last Friday was that when no ack was received, the original message was added to the queue each time ... e.g. after 5 re-send tries, I had 5 times the same message in the queue. Then afterwards ... when the receiving system sent an ack for the first message, it received also the other copies ... (one by one).

In prior versions of Mirth Connect I encountered a similar behaviour when the "Keep Connection Open" property was set to "Yes".

Best Regards

Nico


Imported Comment. Original Details:
Author: nicovn
Created: 2010-09-06T01:59:11.000-0700
